### PR TITLE
Type Annotations for df.boxplot

### PR DESCRIPTION
**Pull request by JojoYY**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Thanks for the PR: Could you please sort your imports using isort
```
4.3.21
Check import format using isort
ERROR: /home/runner/work/pandas/pandas/pandas/plotting/_core.py Imports are incorrectly sorted.
```

File: pandas/plotting/_core.py [hunk 1]
```diff
@@ -391,14 +393,14 @@ def hist_frame(
 @Appender(_boxplot_doc)
 def boxplot(
     data,
-    column=None,
+    column: Optional[Union[str, List[str]]] = None,
     by=None,
```
Comment (alimcmaster1) [inline pandas/plotting/_core.py]: Could we type these too?

**Review by alimcmaster1 — CHANGES_REQUESTED** [review]

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good and I think CI failures are unrelated / have been fixed on master since this was created.

To be sure, can you locally on your `pydata_sprint` branch run:

```sh
git fetch upstream
git merge upstream/master
git push origin pydata_sprint
```

I think should get CI green for this. Flag me down at PyData if you have any questions

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -391,14 +393,14 @@ def hist_frame(
...
-    fontsize=None,
-    rot=0,
-    grid=True,
-    figsize=None,
-    layout=None,
+    fontsize: Optional[Union[float, str]] = None,
+    rot: Optional[Union[float, int]] = 0,
+    grid: bool = True,
+    figsize: Optional[Tuple[int, float]] = None,
+    layout: Optional[Tuple[int]] = None,
```
Comment (simonjayhawkins) [inline pandas/plotting/_core.py]:
```suggestion
    layout: Optional[Tuple[int, int]] = None,
```
`layout`is a 2-tuple of (row,columns)

File: pandas/plotting/_core.py [hunk 3]
```diff
@@ -391,14 +393,14 @@ def hist_frame(
...
+    column: Optional[Union[str, List[str]]] = None,
     by=None,
     ax=None,
-    fontsize=None,
-    rot=0,
-    grid=True,
-    figsize=None,
-    layout=None,
+    fontsize: Optional[Union[float, str]] = None,
+    rot: Optional[Union[float, int]] = 0,
```
Comment (simonjayhawkins) [inline pandas/plotting/_core.py]:
```suggestion
    rot: Optional[float] = 0,
```
int is a subtype of float

File: pandas/plotting/_core.py [hunk 4]
```diff
@@ -391,14 +393,14 @@ def hist_frame(
...
+    grid: bool = True,
+    figsize: Optional[Tuple[int, float]] = None,
```
Comment (simonjayhawkins) [inline pandas/plotting/_core.py]:
```suggestion
    figsize: Optional[Tuple[float, float]] = None,
```
figsize is a 2-tuple (width, height). I assume that if height accepts a float, so does width?

File: pandas/plotting/_core.py [hunk 5]
```diff
@@ -391,14 +393,14 @@ def hist_frame(
 @Appender(_boxplot_doc)
 def boxplot(
     data,
-    column=None,
+    column: Optional[Union[str, List[str]]] = None,
```
Comment (simonjayhawkins) [inline pandas/plotting/_core.py]: from the docstring...
```
column : str or list of str, optional
    Column name or list of names, or vector.
```
is vector any sequence of strings?

**Comment by jreback** [comment]
closing as stale, if you'd like to continue, pls ping.

**Pull request closed without merging**